### ISSUE TITLE
salt 2014.7.5: "TypeError: string indices must be integers, not str" in file.managed

### ISSUE THREAD

**Issue by mschiff** [saltstack/salt#23186]
A state like the following with file.managed won't work and throws an exception:

```yaml
/etc/portage/package.use:
  file.managed:
    - template: jinja
    - makedirs: True
    - source: salt://portage/files/package.use
```

Exception:
```yaml
----------
          ID: /etc/portage/package.use
    Function: file.managed
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib64/python2.7/site-packages/salt/state.py", line 1542, in call
                  **cdata['kwargs'])
                File "/usr/lib64/python2.7/site-packages/salt/states/file.py", line 1354, in managed
                  **kwargs
                File "/usr/lib64/python2.7/site-packages/salt/modules/file.py", line 2877, in check_managed_changes
                  **kwargs)
                File "/usr/lib64/python2.7/site-packages/salt/modules/file.py", line 2521, in get_managed
                  sfn = __salt__['cp.cache_file'](source, saltenv)
                File "/usr/lib64/python2.7/site-packages/salt/modules/cp.py", line 337, in cache_file
                  result = __context__['cp.fileclient'].cache_file(path, saltenv)
                File "/usr/lib64/python2.7/site-packages/salt/fileclient.py", line 147, in cache_file
                  return self.get_url(path, '', True, saltenv)
                File "/usr/lib64/python2.7/site-packages/salt/fileclient.py", line 522, in get_url
                  return self.get_file(url, dest, makedirs, saltenv)
                File "/usr/lib64/python2.7/site-packages/salt/fileclient.py", line 993, in get_file
                  if not data['data']:
              TypeError: string indices must be integers, not str
     Started: 23:21:59.298874
    Duration: 167.252 ms
     Changes:

```

It seems that salt is very picky about the filename you choose as the `source:` value.

If I rename the source file to `package.use.conf` for example and use that in the source statement, everything works fine.

Salt should accept any valid filename as source filename.

**Comment by jfindlay** [comment]
This is most likely a duplicate of #21480.

**Comment by rallytime** [comment]
This has been fixed by #24395 and the 2015.5.3 (or newer) release should have the fix in it. As such, I am going to close this issue. If this pops up again please let us know! Thanks.

**Comment by yarons** [comment]
Ran into it again in 2015.8.6.
